src/locale/localed.c

```diff
@@ -298,13 +299,59 @@ static int process_locale_list_item(
         return sd_bus_error_setf(error, SD_BUS_ERROR_INVALID_ARGS, "Locale assignment %s not valid, refusing.", assignment);
 }
 
+static int locale_gen_process_locale(char *new_locale[static _VARIABLE_LC_MAX],
+                                     sd_bus_error *error) {
+        int r;
+        assert(new_locale);
+
+        for (LocaleVariable p = 0; p < _VARIABLE_LC_MAX; p++) {
+                if (p == VARIABLE_LANGUAGE)
+                        continue;
+                if (isempty(new_locale[p]))
+                        continue;
+                if (locale_is_installed(new_locale[p]))
+                        continue;
+
+                r = locale_gen_enable_locale(new_locale[p]);
+                if (r != 0) {
+                        if (r == -ENOEXEC) {
+                                log_error_errno(r, "Refused to enable locale for generation: %m");
+                                return sd_bus_error_setf(error,
+                                                                SD_BUS_ERROR_INVALID_ARGS,
+                                                                "Specified locale is not installed and non-UTF-8 locale will not be auto-generated: %s",
+                                                                new_locale[p]);
+                        }
+
+                        if (r == -EINVAL) {
+                                log_error("Failed to enable invalid locale %s for generation.", new_locale[p]);
+                                return sd_bus_error_setf(error,
+                                                                SD_BUS_ERROR_INVALID_ARGS,
+                                                                "Can not enable locale generation for invalid locale: %s",
+                                                                new_locale[p]);
+                        }
+
+                        log_error_errno(r, "Failed to enable locale for generation: %m");
+                        return sd_bus_error_set_errnof(error, r, "Failed to enable locale generation: %m");
+                }
+
+                r = locale_gen_run();
+                if (r != 0) {
```

Comment:
functions are supposed to return < 0 on error according to our coding style (>= 0 is for different types of success, if there are multiple), and this is where you handle the error conditions, hence the check should be `if (r < 0)`.

See https://systemd.io/CODING_STYLE/#error-handling